Scripts/utils/validate_network.py

```diff
@@ -95,7 +95,7 @@ def validate(network, fares=None):
                 log.error(msg)
                 raise ValueError(msg)
         if network.mode('t') in link.modes or network.mode('p') in link.modes:
-            speedstr = str(int(link.data1))
+            speedstr = str(int(link.data1)).zfill(6)
             speed = {
                 "aht": int(speedstr[:-4]),
                 "pt": int(speedstr[-4:-2]),
```

Comment:
The reason why `speedstr` is sliced backwards from the end is that the string may be either 5 or 6 digits long. `zfill` is an elegant solution to this that would allow us to do the slicing from start to end.

Reply:
Sorry, not sure I got the point. Do you mean this implementation is now good? Or are you still proposing some changes?

Reply:
@zptro Do you think this branch is ready for merge now?

Reply:
Sorry, I had missed this conversation. I mean that the slicing on rows 100-102 could be done in in a way that is easier to understand (without negative values) now that you have performed `zfill`. Otherwise there is no point in doing `zfill`.